frontend_lib/src/htmlRemoveInteractivity.js

```diff
@@ -0,0 +1,102 @@
+  // children. Replacing <a> by either <span> or <div> won't work.
+  // Unfortunately, there are no transparent elements in the HTML spec that are
+  // suitable for replacing <a>.
+  // A solution is therefore to create a custom element. Custom elements are
+  // supported in all common browsers.
+  // If supported, we define the custom element. Otherwise, we just create the
+  // element as if it existed and it will have the correct behavior.
+
+  // [1] See:
+  // - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a
+  // - https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element
+  // - https://html.spec.whatwg.org/multipage/dom.html#transparent
+
+  if (window.customElements && !window.customElements.get('a-in-preview')) {
+    window.customElements.define('a-in-preview',
+      class extends HTMLElement {}
+    )
+  }
+
+  const fakeLink = doc.createElement('a-in-preview')
+
+  if (href) {
+    fakeLink.setAttribute('title', href)
+  }
+
+  return fakeLink
+}
+
+function maybeReplaceNode (node, doc) {
+  // we remove interactive content, listed here:
+  // https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#Interactive_content
+  // We also remove style and script tags
+
+  if (!(node instanceof Element)) {
+    return node.cloneNode(false)
+  }
+
+  const nodeName = node.nodeName.toLowerCase()
+
+  switch (nodeName) {
+    case 'style':
+    case 'script':
+    case 'input':
+    case 'embed':
+    case 'iframe':
+    case 'keygen':
+    case 'button':
+    case 'menu':
+    case 'select':
+      return null
+    case 'label':
+      return doc.createElement('span')
+    case 'a':
+      return getFakeAnchorElement(doc, node.href)
+    case 'summary':
+      return doc.createElement('div')
+    case 'details':
+      return doc.createElement('strong')
+  }
+
+  return doc.createElement(nodeName)
+}
+
+function removeInteractiveContentFromDOM (node, doc) {
+  const newNode = maybeReplaceNode(node, doc)
+
+  if (!newNode) {
+    return null
+  }
+
+  // remove javascript handlers, attribute that make some elements interactive, anchor name, id.
+  if (node.attributes) {
+    for (const attr of [].slice.call(node.attributes)) {
+      if (!attr.name.startsWith('on') && !ATTRIBUTES_TO_REMOVE.includes(attr.name)) {
+        newNode.setAttribute(attr.name, attr.value)
+      }
+    }
+  }
+
+  for (const child of node.childNodes) {
+    const newChild = removeInteractiveContentFromDOM(child, doc)
+    if (newChild) {
+      newNode.appendChild(newChild)
+    }
+  }
+
+  return newNode
+}
+
+export function removeInteractiveContentFromHTML (html) {
+  // RJ - 2020-11-09 - NOTE
+  // Previews are in a <a> tag. It is forbidden to have interactive
+  // content inside a <a> tag in HTML, so let's remove interactivity from previews.
+  const doc = new DOMParser().parseFromString(html, 'text/html')
+  return removeInteractiveContentFromDOM(doc.body, doc).innerHTML
+}
```

Comment:
Interesting approach, did you consider using a transparent overlay to intercept clicks in order to avoid having to edit all this content (I don't know if it would have been easier, just wondering)?

Reply:
It's not valid HTML, this is why I did this

Reply:
Okay, a originally misinterpreted @grignards's proposal. He was suggesting to add a <a> overlay over the html content to avoid both nesting interactive content in <a> and having to fiddle with the preview's DOM. 

We had a long discussion with @grignards and this is a tricky subject with no good answers.
The current solution has the cons to require DOM manipulations. His is simpler but breaks keyboard navigation and requires to think about the overlay's position.

We are keeping the current solution for now.